src/Resources/Custom TypeEditor Template.md

```diff
@@ -0,0 +1,314 @@
+    to convert the data its hosting into the JSON format and back from it, which
+    is needed to transport it across the processes bounderies. With that, the
+    response class for the endpoint to create the server-side view model not
+    only takes the proxy of the server-side view model, but also the original
+    values of the types, the custom property type is composed of. And this data
+    we now use to populate the type editor’s dialog client side.
+
+-   The user now edits the values.
+
+-   When the user now clicks OK, we validate the data on the client inside the
+    \`CustomTypeEditorDialog\`. And if the validation passes, the dialog’s
+    returns \`DialogResult.OK\`, and we call the \`ExecuteOKCommand\` method of
+    the client view model, to kick of the data transfer process to the server.
+    This method now sends the \`CustomTypeEditorOKClickRequest\` to the server
+    and passes the induvial retrieved data from the user’s input in the dialog
+    along. The endpoint’s handler gets those data and passes - in turn - that
+    data to the server-side view model, which then again calls the \`OnClick\`
+    method, composes the actual instance of the custom control’s property type,
+    and stores it in the \`PropertyStore\` property of the server-side view
+    model. And with that the call chain seems to end here. So, the server-side
+    ViewModel now holds the edited and committed result. The question now is:
+    How does the ViewModel property find the way back to the control’s property?
+    That is done client-side: Remember? When the client-side view model got
+    created, it not only triggered the creation of the server-side view model.
+    It also requested the proxy of that view model to be returned to the client
+    side. On the client, the client-side ViewModel holds the reference to
+    server-side view model’s PropertyStore property over a ProxyObject. When the
+    user clicks OK in the editor, that code flow is returned to the the
+    TypeEditor (running in the context of Visual Studio), which opened the modal
+    dialog to begin with. Now, back in the actual type editor class it is where
+    the assignment from this ViewModel to the actual Property of the Control
+    happens:
+
+var dialogResult = editorService.ShowDialog(_customTypeEditorDialog);
+
+if (dialogResult == DialogResult.OK)
+
+{
+
+// By now, the UI of the Editor has asked its (client-side) ViewModel
+
+// to run the code which updates the property value. It passes the data to
+
+// the server, which in turn updates the server-side ViewModel.
+
+// When it's time to return the value from the client-side ViewModel back to the
+
+// Property Browser (which has called the TypeEditor in the first place), the
+client-side
+
+// ViewModel accesses its PropertyStore property, which in turn gets the
+required PropertyStore
+
+// proxy object directly from the server-side ViewModel.
+
+value = viewModelClient.PropertyStore;
+
+}
```

Comment:
Broken formatting. Code samples must be wrapped with tripple backticks followed by the language code (`cs` for C# or `vb` for Visual Basic).